src/devices/lacrosse_TX141TH_Bv2.c

```diff
@@ -0,0 +1,227 @@
+
+typedef struct {
+    int32_t	data;	// First 4 data bytes compressed into 32-bit integer
+    int8_t count;	// Count
+} data_and_count;
+
+static int lacrosse_tx141th_bv2_callback(bitbuffer_t *bitbuffer) {
+	bitrow_t *bb = bitbuffer->bb;
+	data_t *data;
+	char time_str[LOCAL_TIME_BUFLEN];
+    local_time_str(0, time_str);
+	int i,j,k,nbytes,npacket,kmax;
+    uint8_t id=0,status=0,battery_low=0,test=0,humidity=0,maxcount;
+    uint16_t temp_raw=0;
+    float temp_f=0.0;
+    data_and_count dnc[LACROSSE_TX141TH_PACKETCOUNT] = {0};
+    
+    if (debug_output) {
+        bitbuffer_print(bitbuffer);
+    }
+    
+    npacket=0; // Number of unique packets
+    for(i=0; i<BITBUF_ROWS; ++i) {
+        j=bitbuffer->bits_per_row[i];
+        if(j>=LACROSSE_TX141TH_BITLEN) {
+            nbytes=j/8;
+            for(j=0;j<nbytes;j+=LACROSSE_TX141TH_BYTELEN) {
+                uint32_t *d=(uint32_t *)(bb[i]+j);
+                uint8_t not_found=1;
+                for(k=0;k<npacket;++k) {
+                    if(*d==dnc[k].data) {
+                        ++(dnc[k].count);
+                        not_found=0;
+                        break;
+                    }
+                }
+                if(not_found) {
+                    dnc[npacket].data=*d;
+                    dnc[npacket].count=1;
+                    if(npacket+1<LACROSSE_TX141TH_PACKETCOUNT) ++npacket;
+                }
+            }
+        }
+    }
+ 
+    if (debug_output) {
+        fprintf(stderr, "%d unique packet(s)\n", npacket);
+        for(k=0;k<npacket;++k) {
+            fprintf(stderr, "%08x \t %d \n", dnc[k].data,dnc[k].count);
+        }
+    }
+
+    // Find the most frequent data packet, if necessary
+    kmax=0;
+    maxcount=0;
+    if(npacket>1) {
+        for(k=0;k<npacket;++k) {
+            if(dnc[k].count>maxcount) {
+                maxcount=dnc[k].count;
+                kmax=k;
+            }
+        }
+    }
+    
+    // Unpack the data bytes back to eliminate dependence on the platform endiannes!
+    uint8_t *bytes=(uint8_t*)(&(dnc[kmax].data));
+    id=bytes[0];
+    status=bytes[1];
+    battery_low=(status & 0x80) >> 7;
+    test=(status & 0x40) >> 6;
+    temp_raw=((status & 0x0F) << 8) + bytes[2];
+    temp_f = 9.0*((float)temp_raw)/50.0-58.0; // Temperature in F
+    humidity = bytes[3];
+
+    if (0==id || 0==humidity || humidity > 100 || temp_f < -40.0 || temp_f > 140.0) {
+        if (debug_output) {
+            fprintf(stderr, "LaCrosse TX141TH-Bv2 data error\n");
+        }
+        return 0;
+    }
+
+    data = data_make("time",    "Date and time", DATA_STRING,    time_str,
+                     "temperature", "Temperature in deg F", DATA_FORMAT, "%.2f F", DATA_DOUBLE, temp_f,
+                     "humidity",    "Humidity", DATA_FORMAT, "%u %%", DATA_INT, humidity,
+                     "quality", "Signal quality",   DATA_FORMAT, "%.1f %%", DATA_DOUBLE, 100.0*((float)dnc[kmax].count)/LACROSSE_TX141TH_PACKETCOUNT,
```

Comment:
Well what unit has the quality factor? Right now it is usable for you with this sensor. How do you apply the same to all the others and make sure the quality factor has the same meaning? These are not questions that can be answered easily thus I'd prefer you to remove it for the initial commit. If we can come up with a quality metric that can be used for all decoder then we can add it outside the decoders. Like the timestamp, but right now every decoder reimplements it because I'm to lazy to fix it.